x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/AutoBucket.java

```diff
@@ -90,8 +90,8 @@ public AutoBucket(
         Source source,
         @Param(name = "field", type = { "integer", "long", "double", "date" }) Expression field,
         @Param(name = "buckets", type = { "integer" }) Expression buckets,
-        @Param(name = "from", type = { "integer", "long", "double", "date", "string" }) Expression from,
-        @Param(name = "to", type = { "integer", "long", "double", "date", "string" }) Expression to
+        @Param(name = "from", type = { "integer", "long", "double", "date", "keyword" }) Expression from,
+        @Param(name = "to", type = { "integer", "long", "double", "date", "keyword" }) Expression to
```

Comment:
Why keep keyword when ImplicitCasting is allowed? I would expect the from/to to NOT list keyword.
Separately this impacts UI hence why I'm pinging the Kibana team.

Reply:
String this info is also used externally for docs, better to handle "string" as "keyword" internally.

Reply:
keyword/string is removed from the type list.